### PR TITLE
[change] Update dependencies #203

### LINKED ISSUE #203: [change] Update dependencies
Depends on https://github.com/openwisp/netjsonconfig/issues/189, which has to be done first.

Let's update all the python dependencies and fix the requirements badge (see how it's done in openwisp-utils for a reference).

### PR DESCRIPTION
**Pull request by yashikajotwani12**
Fixes #203

### CODE REVIEW

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Please can you also update the dependency monitoring service as done in https://github.com/openwisp/openwisp-utils/commit/243f119132d4ebcf5e9bd6af2c2dcab635a42b2f?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Please also update the other outdated deps, see https://libraries.io/github/openwisp/netjsonconfig#repository_dependencies.

**Comment by yashikajotwani12** [comment]
> Please can you also update the dependency monitoring service as done in [openwisp/openwisp-utils@243f119](https://github.com/openwisp/openwisp-utils/commit/243f119132d4ebcf5e9bd6af2c2dcab635a42b2f)?

Yeah, sure I will. 👍🏻

**Review by pandafy — CHANGES_REQUESTED** [review]
@yashikajotwani12 the latest version of `jsonschema` is `4.4.0`. We should update that as well. 

https://github.com/openwisp/netjsonconfig/blob/ca737e32436d9216a2dec10813d164acc4aa310e/requirements.txt#L1-L3

@nemesisdesign I think it will be save to remove `six` from the requirements. What do you think?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
> @nemesisdesign I think it will be save to remove `six` from the requirements. What do you think?

Yes we can remove it.

**Review by pandafy — COMMENTED** [review]
`hostname` validation has been [updated in `jsonschema`](https://github.com/Julian/jsonschema/blob/ee5d31afb08ef0a1592ce54431f3f9c59ad95bbf/jsonschema/_format.py#L233-L247).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
 
 jobs:
   build:
-    name: Python==${{ matrix.python-version }}
+    name: Python==${{ matrix.python-version }} | ${{ matrix.django-version }}
```
Comment (pandafy) [inline .github/workflows/ci.yml:14]: @yashikajotwani12 why did you made this change? This module is not dependent on Django.

Comment (yashikajotwani12) [reply .github/workflows/ci.yml]: Maybe I made it by mistake, I will take care next time, Thanks for the information

File: netjsonconfig/backends/base/backend.py [hunk 2]
```diff
@@ -131,6 +134,24 @@ def _cidr_notation(value):
...
+        """
+        if not isinstance(value, str):
+            return True
+        if not _host_name_re.match(value):
+            return False
+        components = value.split(".")
+        for component in components:
+            if len(component) > 63:
+                return False
+        return True
```
Comment (nemesisdesign) [inline netjsonconfig/backends/base/backend.py:153]: great, thanks! :+1:

**Review by nemesisdesign — APPROVED** [review]
Thanks! :+1:

**Pull request merged**